code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -606,10 +607,59 @@ datum/reagent/shadowling_blindness_smoke //Reagent used for above spell
 				return
 
 
+/obj/effect/proc_holder/spell/targeted/shadowling_extend_shuttle
+	name = "Destroy Engines"
+	desc = "Extends the time of the emergency shuttle's arrival by ten to fifteen minutes."
+	panel = "Shadowling Abilities"
+	range = 1
+	clothes_req = 0
+	charge_max = 600
+	action_icon_state = "extend_shuttle"
+
+/obj/effect/proc_holder/spell/targeted/shadowling_extend_shuttle/cast(list/targets, mob/living/carbon/human/U = usr)
+	if(!shadowling_check(usr))
+		charge_counter = charge_max
+		return
+	for(var/mob/living/carbon/human/target in targets)
+		if(!ishuman(target) || !target)
+			charge_counter = charge_max
+			return
+		if(target.stat)
+			charge_counter = charge_max
+			return
+		if(!is_thrall(target))
+			usr << "<span class='warning'>[target] must be a thrall.</span>"
+			charge_counter = charge_max
+			return
+		if(SSshuttle.emergency.mode != SHUTTLE_CALL)
+			usr << "<span class='warning'>The shuttle must be inbound only to the station.</span>"
+			charge_counter = charge_max
+			return
+		var/mob/living/carbon/human/M = target
+		U.visible_message("<span class='warning'>[U]'s eyes flash a bright red!</span>", \
+						  "<span class='notice'>You begin to draw [M]'s life force.</span>")
+		M.visible_message("<span class='warning'>[M]'s face falls slack, their jaw slightly distending.</span>", \
+						  "<span class='boldannounce'>You are suddenly transported... far, far away...</span>")
+		if(!do_after(U, 50, target = M))
+			M << "<span class='warning'>You are snapped back to reality, your haze dissipating!</span>"
+			U << "<span class='warning'>You have been interrupted. The draw has failed.</span>"
+			return
+		U << "<span class='notice'>You project [M]'s life force toward the approaching shuttle, extending its arrival duration!</span>"
+		M.visible_message("<span class='warning'>[M]'s eyes suddenly flare red. They proceed to collapse on the floor, not breathing.</span>", \
+						  "<span class='warning'><b>...speeding by... ...pretty blue glow... ...touch it... ...no glow now... ...no light... ...nothing at all...</span>")
+		M.death()
+		if(SSshuttle.emergency.mode == SHUTTLE_CALL)
+			var/more_minutes = 9000
+			var/timer = SSshuttle.emergency.timeLeft()
+			timer += more_minutes
+			priority_announce("Major system failure aboard the emergency shuttle. This will extend its arrival time by approximately 15 minutes..", "System Failure", 'sound/misc/notice1.ogg')
+			SSshuttle.emergency.setTimer(timer)
```

Comment:
Why? There's a lot of instances with shuttle things being modified outside of just that file.